zap/src/main/java/org/zaproxy/zap/spider/SpiderController.java

```diff
@@ -251,46 +248,71 @@ public void init() {
 
     /** Clears the previous process. */
     public void reset() {
-        visitedGet.clear();
-        visitedPost.clear();
+        visitedResources.clear();
 
         for (SpiderParser parser : parsers) {
             parser.removeSpiderParserListener(this);
         }
     }
 
-    @Override
-    public void resourceURIFound(
-            HttpMessage responseMessage, int depth, String uri, boolean shouldIgnore) {
-        log.debug("New resource found: " + uri);
+    /**
+     * Builds a canonical identifier for found resources considering the method, URI, headers, and
+     * body.
+     *
+     * @param uri uniform resource identifier for resource
+     * @param resourceFound resource found
+     * @return identifier as a string representation usable for equality checks
+     */
+    private String buildCanonicalResourceIdentifier(URI uri, SpiderResourceFound resourceFound)
+            throws URIException {
+        StringBuilder identifierBuilder = new StringBuilder(50);
+        if (resourceFound != null) {
```

Comment:
Is this needed?

Reply:
Removed check in last commit as not needed